onnxruntime/python/tools/quantization/quantize.py

```diff
@@ -522,7 +526,15 @@ def inc_dataloader():
             use_external_data_format=use_external_data_format,
             extra_options=calib_extra_options,
         )
-        calibrator.collect_data(calibration_data_reader)
+
+        if stride:
```

Comment:
There are a few instances of variables not being defined before use. I would recommend running `lintrunner` locally to help catch some of these issues. Here's a guide how to do that: https://github.com/microsoft/onnxruntime/blob/main/docs/Coding_Conventions_and_Standards.md#linting